nncf/common/quantization/initialization/range.py

```diff
@@ -133,3 +134,42 @@ def __init__(self, init_range_data_loader: NNCFDataLoader,
         self.device = device
         self.global_init_config = global_init_config
         self.per_layer_range_init_configs = per_layer_range_init_configs
+
+
+class RangeInitCollectorParams:
+    def __init__(self, is_weights: bool, mode: QuantizationMode, per_channel: bool, init_type: str):
+        self._is_weights = is_weights
+        self._mode = mode
+        self._per_channel = per_channel
+        self._init_type = init_type
+
+    @property
+    def _use_abs_max(self) -> bool:
+        return self._mode == QuantizationMode.SYMMETRIC
+
+    @property
+    def _use_means_of_mins(self) -> bool:
+        return not self._is_weights and not self._per_channel and self._mode == 'asymmetric'
+
+    @property
+    def _use_means_of_maxs(self) -> bool:
+        return not self._is_weights and not self._per_channel
+
+    def _get_params_for_min_max(self) -> bool:
+        return self._use_abs_max
+
+    def _get_params_for_mixed_min_max(self) -> Tuple[bool, bool, bool]:
+        return self._use_abs_max, self._use_means_of_mins, self._use_means_of_maxs
+
+    def _get_params_for_mean_min_max(self) -> bool:
+        return self._use_abs_max
+
+    def get_low_level_params_for_collector(self):
+        """Generate low-level parameters for collectors"""
+        if self._init_type == "min_max":
+            return self._get_params_for_min_max()
+        if self._init_type == "mixed_min_max":
+            return self._get_params_for_mixed_min_max()
+        if self._init_type == "mean_min_max":
+            return self._get_params_for_mean_min_max()
+        raise RuntimeError("Parameters not required or unknown range init type: {}".format(self._init_type))
```

Comment:
This could really stand to be refactored. At the very least the `_init_type` must be made a specific enum instead of an opaque str. A further improvement would be to avoid the functions for which you cannot write a meaningful return type hint such as this one, which essentially returns different object set in different situation. A more deep refactoring would have to be made for the latter to be fixed - you may postpone it by creating a ticket for this.

Reply:
As discused, `get_low_level_params_for_collector()` can be eliminated by crating public properties in `RangeInitCollectorParams`, which will be directly feed to instance collectors.

Reply:
Fixed, thanks!